### DISCUSSION TITLE
srcset for images needed

### DISCUSSION THREAD

**Discussion by no-matter-0** [joomla/joomla-cms#28368] [2. Cold Features]
I find it rather urgent for Joomla 4 to have a srcset for the images...

`<img src=""  srcset="" alt="">`
JCE editor has it already
thanks to the team

**Comment by Quy** [comment]
Is this a duplicate of #25655?

**Comment by no-matter-0** [comment]
not really, I dont want my photos to be resized , just to have the ability to add the to the srcset

**Comment by richard67** [comment]
@no-matter-0 You need that in the editor in backend e.g. for images inserted inside articles? Or also at other places, e.g. intro image of article?

To all who might know: Is that something to be done in TinyMCE, or does TinyMCE already support it and we only don't use it?

**Comment by no-matter-0** [comment]
we need it everywhere, it should be the modern way to display any img

`<img src="/images/photo-1x.jpg" srcset="/images/photo-1x.jpg 1x, /images/photo-2x.jpg 2x, /images/photo-3x.jpg 3x" alt="">`

of course in tinymce it would be great too , actually only JCE support it (since one year I think)

**Comment by N6REJ** [comment]
"JOOMLA!" can't natively support this, since it doesn't have a bulk image insertion, without a major rewrite of com_media.  This should be looked into via a function of the editor. https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images

**Comment by no-matter-0** [comment]
Nowadays it is necessary img src alone is not enough
Tinymce is not a serious editor, I use Codemirror but it is not an editor at all

**Comment by ghost** [comment]
@richard67 @no-matter-0  What benefit is to have many 10+MB photos not able to be resized by Joomla core, but just to have the ability to add the to the srcset?

We need that in the editor in frontend e.g. for images inserted inside articles and we need it everywhereelse also.

**Comment by richard67** [comment]
@Milglius Please don't abuse other peoples' issues to discuss yours.

This issue here is about using SRC set.

You talk about possibility to resize pictures and save with different name on upload.

Both features can be used together of course, but they are different features.

**Comment by ghost** [comment]
@richard67 please dont abuse me.

My comment here is about using SRC set.

**Comment by no-matter-0** [comment]
@Milglius No serious photographer or graphic designer in the world will automatically resize their photos with Joomla. 

 If you don't know this subject why answer?

We need srcset for our already resized images 
And adding this code (this possibility) is not a big job. 
I have my own library that creates
img (src srcset  alt)
and/or
picture (source srcset media sizes) (img src alt) 

Automatic image resizing is another subject that has nothing to do with this one.

**Comment by ghost** [comment]
@no-matter-0  My comment here is about what benefit using SRC set then in joomla core you have many 10+MB photos?

I doubt joomla core should be just for photographer or graphic designer who resize their photos manualy and do not use Joomla,

**Comment by no-matter-0** [comment]
adding a srcset attributes is nothing to do , and as i said it is already in JCE  (without resizing) since at least one year

resizing images is another subject , feel free to open it if you think it is important

thanks for your comments

**Comment by ghost** [comment]
@no-matter-0 how you adding a srcset in JCE (without resizing) ?

**Comment by no-matter-0** [comment]
in JCE there is  a possibility to add a row for each images you want to ad to the srcset, you choose your image just as you choose the one for src , and of course you can fill attributes like density (2x)  or sizes (400w)

Why worry about resizing ?

**Comment by richard67** [comment]
@Milglius Please respect if the author of this issue tells you that in his opinion your discussion is not related to his issue, and don't force him into your discussion by posting again and again.

**Comment by no-matter-0** [comment]
It could be related  and certainly useful for many , but not necessary
JCE is a great editor and implements srcset without resizing

thank you for reading me

**Comment by ghost** [comment]
@no-matter-0 can you show us JCE printscr with that possibility - to add a row for each images you want to ad to the srcset, you choose your image just as you choose the one for src , and of course you can fill attributes like density (2x) or sizes (400w)

**Comment by no-matter-0** [comment]
1) go to https://demo.joomlacontenteditor.net/  login with demo / demo
2) click any edit box in front end,
3) click on the second  insert image icon (the one for responsive images)
4) go to responsive tab
5) source row with SRC W X  (optional sizes attribute)
6) PLUS to add another row

easy simple and clever

JCE has now the option for images resizing but the firsts versions with srcset was without it

**Comment by no-matter-0** [comment]
you can try it yourself and save and you wil get something like

`<img src="/images/sampledata/dandelion.jpg" srcset="/images/sampledata/cockroach.jpg 400w/1x,images/sampledata/owl.jpg 800w/2x" alt="dandelion" width="1920" height="1151">`

**Comment by N6REJ** [comment]
> We need that in the editor in frontend e.g. for images inserted inside articles and we need it everywhereelse also.

has nothing to do with srcset.
